system/jlib/jmetrics.cpp

```diff
@@ -68,7 +69,11 @@ void MetricsReporter::addMetric(const std::shared_ptr<IMetric> &pMetric)
     }
     else
     {
-        throw MakeStringException(MSGAUD_operator, "addMetric - Attempted to add duplicate named metric with name %s", pMetric->queryName().c_str());
+        //If there is a match only report an error if the metric has not been destroyed in the meantime
+        auto match = it->second.lock();
+        if (match)
+            OERRLOG("addMetric - Adding a duplicate named metric '%s', old metric replaced", pMetric->queryName().c_str());
+        it->second = pMetric;
```

Comment:
I worry that erroneously duplicate named metrics will go undetected. If the instrumented code has made an error and duplicated a metric name, it would replace the existing.

Reply:
It is a trade off.  You can either have the error less likely to be spotted, but allow a system to continue, or make sure the error is reported and cause a system to fail.  On balance we decided keeping going is probably more valuable.

Reply:
One alternative would be to report an error in debug, and continue in release.  I think I will modify to do that.

Reply:
That would achieve a good balance.